lerobot/common/robot_devices/motors/dynamixel.py

```diff
@@ -0,0 +1,588 @@
+    "x_series": X_SERIES_CONTROL_TABLE,
+    "xl330-m077": X_SERIES_CONTROL_TABLE,
+    "xl330-m288": X_SERIES_CONTROL_TABLE,
+    "xl430-w250": X_SERIES_CONTROL_TABLE,
+    "xm430-w350": X_SERIES_CONTROL_TABLE,
+    "xm540-w270": X_SERIES_CONTROL_TABLE,
+}
+
+
+def uint32_to_int32(values: np.ndarray):
+    """
+    Convert an unsigned 32-bit integer array to a signed 32-bit integer array.
+    """
+    for i in range(len(values)):
+        if values[i] is not None and values[i] > 2147483647:
+            values[i] = values[i] - 4294967296
+    return values
+
+
+def int32_to_uint32(values: np.ndarray):
+    """
+    Convert a signed 32-bit integer array to an unsigned 32-bit integer array.
+    """
+    for i in range(len(values)):
+        if values[i] is not None and values[i] < 0:
+            values[i] = values[i] + 4294967296
+    return values
+
+
+def motor_position_to_angle(position: np.ndarray) -> np.ndarray:
+    """
+    Convert from motor position in [-2048, 2048] to radian in [-pi, pi]
+    """
+    return (position / 2048) * 3.14
+
+
+def motor_angle_to_position(angle: np.ndarray) -> np.ndarray:
+    """
+    Convert from radian in [-pi, pi] to motor position in [-2048, 2048]
+    """
+    return ((angle / 3.14) * 2048).astype(np.int64)
+
+
+# def pwm2vel(pwm: np.ndarray) -> np.ndarray:
+#     """
+#     :param pwm: numpy array of pwm/s joint velocities
+#     :return: numpy array of rad/s joint velocities
+#     """
+#     return pwm * 3.14 / 2048
+
+
+# def vel2pwm(vel: np.ndarray) -> np.ndarray:
+#     """
+#     :param vel: numpy array of rad/s joint velocities
+#     :return: numpy array of pwm/s joint velocities
+#     """
+#     return (vel * 2048 / 3.14).astype(np.int64)
+
+
+def get_group_sync_key(data_name, motor_names):
+    group_key = f"{data_name}_" + "_".join(motor_names)
+    return group_key
+
+def get_result_name(fn_name, data_name, motor_names):
+    group_key = get_group_sync_key(data_name, motor_names)
+    rslt_name = f"{fn_name}_{group_key}"
+    return rslt_name
+
+def get_queue_name(fn_name, data_name, motor_names):
+    group_key = get_group_sync_key(data_name, motor_names)
+    queue_name = f"{fn_name}_{group_key}"
+    return queue_name
+
+def get_log_name(var_name, fn_name, data_name, motor_names):
+    group_key = get_group_sync_key(data_name, motor_names)
+    log_name = f"{var_name}_{fn_name}_{group_key}"
+    return log_name
+
+class TorqueMode(enum.Enum):
+    ENABLED = 1
+    DISABLED = 0
+
+
+class OperatingMode(enum.Enum):
+    VELOCITY = 1
+    POSITION = 3
+    EXTENDED_POSITION = 4
+    CURRENT_CONTROLLED_POSITION = 5
+    PWM = 16
+    UNKNOWN = -1
+
+
+class DriveMode(enum.Enum):
+    NON_INVERTED = 0
+    INVERTED = 1
+
+
+class DynamixelMotorsBus:
+    """
+    Example of usage for 1 motor connected to the bus:
+    ```python
+    motor_name = "gripper"
+    motor_index = 6
+    motor_model = "xl330-m077"
+
+    robot = DynamixelMotorsBus(
+        port="/dev/tty.usbmodem575E0031751",
```

Comment:
Where do you get this address from?

Reply:
Added documentation and script

    A DynamixelMotorsBus instance requires a port (e.g. `DynamixelMotorsBus(port="/dev/tty.usbmodem575E0031751"`)).
    To find the port, you can run our utility script:
    ```bash
    python lerobot/common/robot_devices/motors/dynamixel.py
    >>> Finding all available ports for the DynamixelMotorsBus.
    >>> ['/dev/tty.usbmodem575E0032081', '/dev/tty.usbmodem575E0031751']
    >>> Remove the usb cable from your DynamixelMotorsBus and press Enter when done.
    >>> The port of this DynamixelMotorsBus is /dev/tty.usbmodem575E0031751.
    >>> Reconnect the usb cable.
    ```
    To find the motor indices, use [DynamixelWizzard2](https://emanual.robotis.com/docs/en/software/dynamixel/dynamixel_wizard2).